projects/datetime-picker/src/lib/datepicker-input.ts

```diff
@@ -74,8 +74,10 @@ export class NgxMatDatepickerInput<D>
   }
   set min(value: D | null) {
     const validValue = this._dateAdapter.getValidDateOrNull(this._dateAdapter.deserialize(value));
+    const valueTimestamp = new Date(validValue as Date | string).getTime();
+    const minTimestamp = new Date(this._min as Date | string).getTime();
```

Comment:
What if the user uses Moment for example? The casting won't work and will fail, that's why everything should go through the date adapter.